in_someip: Initial input plugin documentation.

anthonypayne-GM opened the pull request:
Adds the documentation for the new in_someip plugin:
  https://github.com/fluent/fluent-bit/pull/9570

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added a documentation page for the SOME/IP input plugin: overview, supported configuration options (Event and RPC), step-by-step getting‑started examples for CLI and config-file usage (YAML and .conf), and a testing walkthrough with a sample test service, representative startup logs, and example received-event payloads (including base64 examples).

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/1569?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

anthonypayne-GM committed:
in_someip: Initial input plugin documentation.

Adds the documentation for the new in_someip plugin:
  https://github.com/fluent/fluent-bit/pull/9570

Signed-off-by: Anthony Payne <anthony.payne@gm.com>

anthonypayne-GM committed:
Merge branch 'fluent:master' into someip-input-plugin-docs

anthonypayne-GM committed:
in_someip: Added yaml sample configuration

Updated in_someip documentation to include example yaml
configuration.

Signed-off-by: Anthony Payne <anthony.payne@gm.com>

eschabell commented:
@esmerel can you review for style, this is waiting on code PR https://github.com/fluent/fluent-bit/pull/9570 before we merge this and is tagged so.

esmerel commented on pipeline/inputs/someip.md:
```diff
@@ -0,0 +1,115 @@
+| Key          | Description |
+| ------------ | ----------- |
+| Event        | SOME/IP event to subscribe to. The configuration can have multiple events, one on each line. An event is identified by a comma separated list with, "_service ID_, _event ID_, _event group ID 1_, _event group ID 2_, ...". The Event must include at least one _event group ID_, but can be associated with multiple. |
+| RPC          | SOME/IP request to send when service is available. The configuration can have multiple RPCs, one on each line. An RPC is composed as a comma separated list with, "_service ID_, _service instance_, _method ID_, _request payload_". The request payload should be a should be base64 encoded |
```
```suggestion
| `Event`       | SOME/IP event to subscribe to. The configuration can have multiple events, one on each line. An event is identified by a comma separated list with, `service_ID, event_ID, event_group_ID_1, event_group_ID_2, ...`. The event must include at least one `event_group_ID`, but can be associated with multiple. |
| `RPC`         | SOME/IP request to send when service is available. The configuration can have multiple RPCs, one on each line. An RPC is composed as a comma separated list with, `service_ID, service_instance, method_ID, request_payload". The request payload should be base64 encoded. |
```
Stylistic consistency

esmerel commented on pipeline/inputs/someip.md:
```diff
@@ -0,0 +1,115 @@
+
+## Getting Started
+
+In order to subscribe to SOME/IP events or send request/receive SOME/IP response, you can run the plugin from the command line or through the configuration file:
```
```suggestion
## Get Started

To subscribe to SOME/IP events or send request/receive SOME/IP response, run the plugin from the command line or through the configuration file:
```

esmerel commented on pipeline/inputs/someip.md:
````diff
@@ -0,0 +1,115 @@
+
+```bash
+$ ./fluent-bit -i someip -p Event=4,1,32768,1 -o stdout
+```
````
```suggestion
### Command line

The _someip_ plugin can be enabled with options from the command line:

```bash
./fluent-bit -i someip -p Event=4,1,32768,1 -o stdout
```
```

esmerel reviewed (commented):
Updated (mostly) for style and consistency. It's not exactly how I want it because the web editor doesn't handle suggested changes with code ticks well.

eschabell committed:
Update pipeline/inputs/someip.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/inputs/someip.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/inputs/someip.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/inputs/someip.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/inputs/someip.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/inputs/someip.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/inputs/someip.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/inputs/someip.md:
```diff
@@ -0,0 +1,115 @@
+
+The plugin supports the following configuration parameters:
+
+| Key          | Description |
+| ------------ | ----------- |
+| Event        | SOME/IP event to subscribe to. The configuration can have multiple events, one on each line. An event is identified by a comma separated list with, "_service ID_, _event ID_, _event group ID 1_, _event group ID 2_, ...". The Event must include at least one _event group ID_, but can be associated with multiple. |
+| RPC          | SOME/IP request to send when service is available. The configuration can have multiple RPCs, one on each line. An RPC is composed as a comma separated list with, "_service ID_, _service instance_, _method ID_, _request payload_". The request payload should be a should be base64 encoded |
+
```
Fixing.

eschabell commented on pipeline/inputs/someip.md:
```diff
@@ -0,0 +1,115 @@
+| Key          | Description |
+| ------------ | ----------- |
+| Event        | SOME/IP event to subscribe to. The configuration can have multiple events, one on each line. An event is identified by a comma separated list with, "_service ID_, _event ID_, _event group ID 1_, _event group ID 2_, ...". The Event must include at least one _event group ID_, but can be associated with multiple. |
+| RPC          | SOME/IP request to send when service is available. The configuration can have multiple RPCs, one on each line. An RPC is composed as a comma separated list with, "_service ID_, _service instance_, _method ID_, _request payload_". The request payload should be a should be base64 encoded |
```
Fixing.

eschabell committed:
docs: pipeline: inputs: someip: fix formatting and markup errors

  - Change all bash code fence identifiers to shell
  - Fix unclosed code fence in command line section
  - Lowercase YAML config keys in example and parameter table
  - Convert duplicate YAML event keys to a proper list
  - Add missing closing code fence for fluent-bit.yaml tab
  - Add missing endtab tag for fluent-bit.yaml tab
  - Add missing closing code fence for fluent-bit.conf tab
  - Fix mismatched closing quote in RPC parameter description
  - Remove trailing space after someip_test_service command
  - Remove extra trailing blank lines at end of file

  Applies to #1569

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
docs: pipeline: inputs: someip: fix formatting, markup, and vale issues

  - Change all bash code fence identifiers to shell
  - Fix unclosed code fence in command line section
  - Add missing closing code fence for fluent-bit.yaml tab
  - Add missing endtab tag for fluent-bit.yaml tab
  - Add missing closing code fence for fluent-bit.conf tab
  - Lowercase YAML config keys in example and parameter table
  - Convert duplicate YAML event keys to a proper YAML list
  - Fix mismatched closing quote in rpc parameter description
  - Replace RPCs with RPC entries to avoid vale spelling flag
  - Fix headings to sentence case
  - Fixing vale compliance errors
  - Remove first-person we usage in Testing section
  - Remove trailing spaces extra trailing blank lines at end of file

  Applies to #1569

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/inputs/someip.md:
```diff
@@ -0,0 +1,113 @@
+| Key          | Description |
+| ------------ | ----------- |
+| `event`       | `SOME/IP` event to subscribe to. The configuration can have multiple events, one on each line. An event is identified by a comma separated list with, `service_ID, event_ID, event_group_ID_1, event_group_ID_2, ...`. The event must include at least one `event_group_ID`, but can be associated with multiple. |
+| `rpc`         | `SOME/IP` request to send when service is available. The configuration can have multiple RPC entries, one on each line. An RPC is composed as a comma separated list with, `service_ID, service_instance, method_ID, request_payload`. The request payload should be base64 encoded. |
```
fixing.

eschabell commented on pipeline/inputs/someip.md:
````diff
@@ -0,0 +1,113 @@
+Sent notification for service 4, event 32768
+```
+
+The following output displays in [Fluent Bit](http://fluentbit.io):
````
Fixing.

eschabell committed:
docs: pipeline: inputs: someip: moree vale compliance fixes

  - Change "comma separated list" to "comma-separated list" in event and rpc descriptions
  - Fix insecure http://fluentbit.io link to https://fluentbit.io

  Applies to #1569

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell reviewed (approved):
@anthonypayne-GM just waiting now on the code PR merging.
